.github/workflows/rust.yml

```diff
@@ -0,0 +1,80 @@
+name: Rust
+
+on:
+  push:
+    branches-ignore:
+      - trying.tmp
+      - staging.tmp
+  pull_request:
+
+jobs:
+  miri:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - env:
+        TARGET: x86_64-unknown-linux-gnu
+      run: sh ci/miri.sh
+  rustfmt_clippy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - env:
+        TARGET: i586-unknown-linux-gnu
+      run: sh ci/tools.sh
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ matrix.channel }}
+        override: true
+    - env:
+        TARGET: ${{ matrix.target }}
+        CHANNEL: ${{ matrix.channel }}
+        CROSS: ${{ matrix.target != 'x86_64-unknown-linux-gnu' && '1' || '0' }}
+        NO_STD: ${{ matrix.target == 'thumbv6m-none-eabi' && '1' || '0' }}
+      run: sh ci/run.sh
+    strategy:
+      matrix:
+        target: [
+          x86_64-unknown-linux-gnu,
+          i686-unknown-linux-gnu,
+          i586-unknown-linux-gnu,
+          armv7-unknown-linux-gnueabihf,
+          aarch64-unknown-linux-gnu,
+          thumbv6m-none-eabi,
+          x86_64-pc-windows-gnu,
+        ]
+        channel: [1.49.0, stable, beta, nightly]
+  test-osx:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ matrix.channel }}
+        override: true
+    - env:
+        TARGET: x86_64-apple-darwin
+        CHANNEL: ${{ matrix.channel }}
+      run: sh ci/run.sh
+    strategy:
+      matrix:
+        channel: [1.49.0, stable, beta, nightly]
+  test-msvc:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ matrix.channel }}
+        override: true
+    - env:
+        TARGET: x86_64-pc-windows-msvc
+        CHANNEL: ${{ matrix.channel }}
+      run: sh ci/run.sh
+    strategy:
+      matrix:
+        channel: [1.49.0, stable, beta, nightly]
```

Comment:
Consider using matrix for the runner/runs-on variable too in order to reduce duplication. I find `strategy.matrix.include` and `strategy.matrix.exclude` to be useful in these situations: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-new-combinations